Mapa, UI z edytorem i kontrolą nad prędkością symulacji

MariuszAKS opened the pull request.

MariuszAKS committed:
First UI prototype

Dodanie panelu manipulującego przebiegiem czasu symulacji:
 - przycisk play/pause
 - 6 przycisków prędkości symulacji (x1, x2, x4, x8, x16, x32)

Dodanie panelu podglądu osobnika

Dodanie placeholdera dla organizmów i wyświetlanie statystyk w podglądzie po naciśnięciu na niego

MariuszAKS committed:
Dodano przykładową mapkę i zmiany widoków

Dodatkowo jest proste generowanie roślin w sektorach, bazując na ich poziomie wegetacji (współpracuje ze zmianą prędkości).

Widoki są jak na razie cztery: trudność poruszania, wegetacja, temperatura i default.

Sektory jak na razie otrzymują wartości zupełnie randomowe dla swoich parametrów.

MariuszAKS committed:
Changed version to the one used in main

MariuszAKS committed:
UI Update (changing values using scrollbar)

Scrollbar for changing values on map (I still have to add changing it to a value of a tile after clicking on one).

Remade scripts to be more independent (using Scriptable Objects).

Few more minor changes in code.

MariuszAKS committed:
Added map editor

Three modes: pointer, brush, bucket
Brush and bucket have two modes and two shapes each, and two described sliders each.
Brush: mode, shape, size, value
Bucket: mode, shape, threshold, value

Still no images for buttons in map editor, and have to add updating labels next to sliders.
Other than that, it's mostly done.

MariuszAKS committed:
Finished editor

Added images
Added updated text next to sliders
Disabled (for now) organism overview panel

MariuszAKS committed:
Merge branch 'main' into UI

MariuszAKS committed:
Changed UI elements placement to anchors

Made it so the UI would generate properly on most resolutions (hopefully)

MariuszAKS committed:
Added controlling the Camera

Added Camera Movement using Arrows and Shift.
Added Zooming in and out functionality.
Added section about camera in MapEditor Pointer.

Also changed map size to be bigger initially.

MariuszAKS committed:
Fixed the error with info about movement

Fixed the error with info about movement staying on screen while changing to brush and bucket options.

MariuszAKS committed:
Fixed: info about movement

Information about movement was clipping out of its panel.

Michal-Roman0 reviewed (approved):
W aspectie 16:9 u mnie nie mieści się słowo "threshold" z wiadra, ale możliwe że to tylko u mnie. Tak czy siak, kawał dobrej roboty.

MariuszAKS commented:
Problemem jest to, że Unity daje zakres czcionki po zaznaczeniu automatycznego skalowania i minimalna jest 18, i jest ona wciąż zbyt duża w tym przypadku, już poprawiam

MariuszAKS committed:
Changed min font size

Changed minimal font size of UI elements to avoid cutting labels (from 18 to 8)
